xarray/tests/test_dataset.py

```diff
@@ -2046,6 +2059,31 @@ def test_expand_dims(self):
         roundtripped = actual.squeeze('z')
         assert_identical(original, roundtripped)
 
+        # Test expanding one dimension to have size > 1 that doesn't have
+        # coordinates, and also expanding another dimension to have size > 1
+        # that DOES have coordinates.
+        actual = original.expand_dims({"d": 4, "e": ["l", "m", "n"]})
```

Comment:
Probably we may need to use an OrderedDict also here for python 3.5 compatibility